Application/EdFi.Ods.AdminApi/Features/ClaimSets/ClaimSetValidator.cs

```diff
@@ -0,0 +1,216 @@
+// SPDX-License-Identifier: Apache-2.0
+// Licensed to the Ed-Fi Alliance under one or more agreements.
+// The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+// See the LICENSE and NOTICES files in the project root for more information.
+
+using EdFi.Ods.AdminApi.Infrastructure.ClaimSetEditor;
+using EdFi.Ods.AdminApi.Infrastructure.Database.Queries;
+using EdFi.Ods.AdminApi.Infrastructure.ErrorHandling;
+using FluentValidation;
+
+namespace EdFi.Ods.AdminApi.Features.ClaimSets;
+
+public class AddClaimSetValidator : AbstractValidator<AddClaimSetRequest>
+{
+    private readonly IGetAllClaimSetsQuery _getAllClaimSetsQuery;
+
+    public AddClaimSetValidator(IGetAllClaimSetsQuery getAllClaimSetsQuery,
+        IGetResourceClaimsAsFlatListQuery getResourceClaimsAsFlatListQuery,
+        IGetAllAuthorizationStrategiesQuery getAllAuthorizationStrategiesQuery)
+    {
+        _getAllClaimSetsQuery = getAllClaimSetsQuery;
+
+        var resourceClaims = (Lookup<string, ResourceClaim>)getResourceClaimsAsFlatListQuery.Execute()
+            .ToLookup(rc => rc.Name?.ToLower());
+
+        var authStrategyNames = getAllAuthorizationStrategiesQuery.Execute()
+            .Select(a => a.AuthStrategyName).ToList();
+
+        RuleFor(m => m.Name).NotEmpty()
+            .Must(BeAUniqueName)
+            .WithMessage(FeatureConstants.ClaimSetAlreadyExistsMessage);
+
+        RuleFor(m => m.Name)
+            .MaximumLength(255)
+            .WithMessage(FeatureConstants.ClaimSetNameMaxLengthMessage);
+
+        RuleFor(m => m).Custom((claimSet, context) =>
+        {
+            var resourceClaimValidator = new ResourceClaimValidator();
+
+            if (claimSet.ResourceClaims != null && claimSet.ResourceClaims.Any())
+            {
+                foreach (var resourceClaim in claimSet.ResourceClaims)
+                {
+                    resourceClaimValidator.Validate(resourceClaims, authStrategyNames,
+                        resourceClaim, claimSet.ResourceClaims, context, claimSet.Name);
+                }
+            }
+        });
+    }
+
+    private bool BeAUniqueName(string? name)
+    {
+        return _getAllClaimSetsQuery.Execute().All(x => x.Name != name);

```

Comment:
== and != always compares case-sensitive.  Set a way ignoring the case of the strings being compared

Reply:
The file has been removed because it's not necessary